File: chor-extraction/src/main/java/ast/sp/nodes/ProcedureInvocationSP.java
package ast.sp.nodes;

import ast.sp.interfaces.Behaviour;

public class ProcedureInvocationSP implements Behaviour {
    private final String procedure;

    public boolean isVisited() {
        return isVisited;
    }

    public void setVisited(boolean visited) {
        isVisited = visited;
    }

    private  boolean isVisited;

    private ProcedureDefinitionSP procedureDefinition;


    public ProcedureInvocationSP(String procedure) {
        this.procedure = procedure;
        isVisited = false;
    }

    public String getProcedure() {
        return procedure;
    }

    public String toString() {
        return procedure + ifVisited();
    }

    public ProcedureDefinitionSP getProcedureDefinition() {
        return procedureDefinition;
    }

    public void setProcedureDefinition(ProcedureDefinitionSP procedureDefinition) {
        this.procedureDefinition = procedureDefinition;
    }

    @Override
    public boolean findRecProcCall(String procname) {
        return procedure.equals(procname);
    }

    private String ifVisited(){
        if (isVisited) return "*";
        else return "";
    }
}

File: chor-extraction/src/main/java/ast/sp/nodes/ProcedureInvocationSP.kt
package ast.sp.nodes

import ast.sp.interfaces.Behaviour

data class ProcedureInvocationSP(val procedure: String) : Behaviour {

    var isVisited: Boolean
    init { isVisited = false }

    override fun toString(): String { return procedure + ifVisited() }

    override fun findRecProcCall(procname: String): Boolean {
        return procedure == procname
    }

    private fun ifVisited(): String {
        if (isVisited)
            return "*"
        else
            return ""
    }

    override fun copy(): Behaviour {
        val prin = ProcedureInvocationSP(procedure)
        prin.isVisited = this.isVisited
        return prin
    }
}
